native/common/jp_convert.cpp

```diff
@@ -82,155 +82,361 @@ class Convert
-				case 's': return &Convert<uint16_t>::toS;
-				case 'i': return &Convert<uint16_t>::toI;
-				case 'j': return &Convert<uint16_t>::toJ;
-				case 'f': return &Convert<uint16_t>::toF;
-				case 'd': return &Convert<uint16_t>::toD;
-			}
-			return 0;
-		case 'i':
-		case 'l':
-			switch (to[0])
-			{
-				case 'z': return &Convert<int32_t>::toZ;
-				case 'b': return &Convert<int32_t>::toB;
-				case 'c': return &Convert<int32_t>::toC;
-				case 's': return &Convert<int32_t>::toS;
-				case 'i': return &Convert<int32_t>::toI;
-				case 'j': return &Convert<int32_t>::toJ;
-				case 'f': return &Convert<int32_t>::toF;
-				case 'd': return &Convert<int32_t>::toD;
-			}
-			return 0;
-		case 'I':
-		case 'L':
-			switch (to[0])
-			{
-				case 'z': return &Convert<uint32_t>::toZ;
-				case 'b': return &Convert<uint32_t>::toB;
-				case 'c': return &Convert<uint32_t>::toC;
-				case 's': return &Convert<uint32_t>::toS;
-				case 'i': return &Convert<uint32_t>::toI;
-				case 'j': return &Convert<uint32_t>::toJ;
-				case 'f': return &Convert<uint32_t>::toF;
-				case 'd': return &Convert<uint32_t>::toD;
-			}
-			return 0;
-		case 'q':
-			switch (to[0])
-			{
-				case 'z': return &Convert<int64_t>::toZ;
-				case 'b': return &Convert<int64_t>::toB;
-				case 'c': return &Convert<int64_t>::toC;
-				case 's': return &Convert<int64_t>::toS;
-				case 'i': return &Convert<int64_t>::toI;
-				case 'j': return &Convert<int64_t>::toJ;
-				case 'f': return &Convert<int64_t>::toF;
-				case 'd': return &Convert<int64_t>::toD;
-			}
-			return 0;
-		case 'Q':
-			switch (to[0])
-			{
-				case 'z': return &Convert<uint64_t>::toZ;
-				case 'b': return &Convert<uint64_t>::toB;
-				case 'c': return &Convert<uint64_t>::toC;
-				case 's': return &Convert<uint64_t>::toS;
-				case 'i': return &Convert<uint64_t>::toI;
-				case 'j': return &Convert<uint64_t>::toJ;
-				case 'f': return &Convert<uint64_t>::toF;
-				case 'd': return &Convert<uint64_t>::toD;
-			}
-			return 0;
-		case 'f':
-			switch (to[0])
-			{
-				case 'z': return &Convert<float>::toZ;
-				case 'b': return &Convert<float>::toB;
-				case 'c': return &Convert<float>::toC;
-				case 's': return &Convert<float>::toS;
-				case 'i': return &Convert<float>::toI;
-				case 'j': return &Convert<float>::toJ;
-				case 'f': return &Convert<float>::toF;
-				case 'd': return &Convert<float>::toD;
-			}
-			return 0;
-		case 'd':
-			switch (to[0])
-			{
-				case 'z': return &Convert<double>::toZ;
-				case 'b': return &Convert<double>::toB;
-				case 'c': return &Convert<double>::toC;
-				case 's': return &Convert<double>::toS;
-				case 'i': return &Convert<double>::toI;
-				case 'j': return &Convert<double>::toJ;
-				case 'f': return &Convert<double>::toF;
-				case 'd': return &Convert<double>::toD;
-			}
-			return 0;
-		case 'n':
-		case 'N':
-		case 'P':
-		default: return 0;
-	}
+
+jconverter getConverter(const char *from, int itemsize, const char *to) {
```

Comment:
Note that ``itemsize`` is now unused.

I chose not to break ABI - however, every use of ``getConverter`` hard-codes the ``to`` argument, so I wonder if we just returned a ``Converter`` instance and accessed the appropriate method (e.g. ``toJ``) if the whole code would become a bit simpler (fewer switches).

In the interests of getting this PR in, I don't think we need to do that here though.